aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -103,6 +103,15 @@ void sigmoid_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void sinc_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.common_dtype(), "sinc_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      scalar_t product = scalar_t(M_PI) * a;
```

Comment:
Same as with CPU kernel: perform the check for zero before computing the product (since the product computation is not necessary if the input is zero)